### PR TITLE
Add `pub` keyword

### PR DESCRIPTION
**Pull request by YerinAlexey**
### Description

This PR adds a new keyword, `pub`, for function declarations that makes them visible to the outside world. Functions are not public by default

### ToDo

- [x] Proposed feature/fix is sufficiently tested
- [x] Proposed feature/fix is sufficiently documented
- [x] The "Unreleased" section in the changelog has been updated, if applicable

### CODE REVIEW

File: docs/developers/specification.md [hunk 1]
```diff
@@ -125,6 +125,7 @@ break
 continue
 else
 false
+pub
```
Comment (garritfra) [inline docs/developers/specification.md:128]: Could you sort this alphabetically?

File: docs/concepts/functions.md [hunk 2]
```diff
@@ -52,7 +54,7 @@ fn add_one(x: int): int {
 Now, if you call the function with `1` as its argument and read its value, you will see the computed result:
 
 ```
-fn main() {
+pub fn main() {
```
Comment (garritfra) [inline docs/concepts/functions.md:57]: Maybe `main` should always be public?

**Review by garritfra — CHANGES_REQUESTED** [review]

File: docs/developers/specification.md (hunk 1 above)
Comment (YerinAlexey) [reply docs/developers/specification.md:128]: This list was not sorted alphabetically before, so I thought this should not be the case

File: docs/concepts/functions.md (hunk 2 above)
Comment (YerinAlexey) [reply docs/concepts/functions.md:57]: Yes

File: docs/developers/specification.md (hunk 1 above)
Comment (garritfra) [reply docs/developers/specification.md:128]: Is it not?

**Comment by garritfra** [comment]
I was thinking about if this should actually be part of the language at this stage. We don't have namespaces yet, so having a `pub` keyword won't do anything. I'll close this PR for now, but we can reference it later, once we have proper namespaces.

**Pull request closed without merging**